src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/ValidateAnomalyDetectorActionHandler.java

```diff
@@ -0,0 +1,590 @@
+    ) {
+        super(client, channel);
+        this.anomalyDetectionIndices = anomalyDetectionIndices;
+        this.anomalyDetector = anomalyDetector;
+        this.maxAnomalyDetectors = maxAnomalyDetectors;
+        this.maxAnomalyFeatures = maxAnomalyFeatures;
+        this.requestTimeout = requestTimeout;
+        this.responseValidate = new ValidateResponse();
+        this.xContent = xContentRegistry;
+        this.inferringInterval = false;
+        this.inferAgain = new AtomicBoolean(true);
+        this.failuresMap = new HashMap<>();
+        this.suggestedChangesMap = new HashMap<>();
+    }
+
+    /**
+     * Start function to process validate anomaly detector request.
+     * Checks if anomaly detector index exist first, if not, add it as a failure case.
+     *
+     * @throws IOException IOException from {@link AnomalyDetectionIndices#initAnomalyDetectorIndexIfAbsent(ActionListener)}
+     */
+    public void startValidation() throws IOException {
+        if (!anomalyDetectionIndices.doesAnomalyDetectorIndexExist()) {
+            anomalyDetectionIndices
+                .initAnomalyDetectorIndex(
+                    ActionListener.wrap(response -> onCreateMappingsResponse(response), exception -> onFailure(exception))
+                );
+        } else {
+            preDataValidationSteps();
+        }
+    }
+
+    public void preDataValidationSteps() {
+        List<String> missingFields = new ArrayList<>();
+        List<String> formatErrors = new ArrayList<>();
+        if (anomalyDetector.getName() == null || anomalyDetector.getName() == "") {
+            missingFields.add("name");
+        } else if (!anomalyDetector.getName().matches(NAME_REGEX)) {
+            formatErrors.add(anomalyDetector.getName());
+            failuresMap.put("format", formatErrors);
+        }
+        if (anomalyDetector.getTimeField() == null) {
+            missingFields.add("time_field");
+        }
+        if (anomalyDetector.getIndices() == null) {
+            missingFields.add("indices");
+        }
+        if (anomalyDetector.getWindowDelay() == null) {
+            missingFields.add("window_delay");
+        }
+        if (anomalyDetector.getDetectionInterval() == null) {
+            missingFields.add("detector_interval");
+        }
+        if (anomalyDetector.getFeatureAttributes().isEmpty()) {
+            missingFields.add("feature_attributes");
+        }
+        if (!missingFields.isEmpty()) {
+            failuresMap.put("missing", missingFields);
+        }
+        String error = RestHandlerUtils.validateAnomalyDetector(anomalyDetector, maxAnomalyFeatures);
+        if (StringUtils.isNotBlank(error)) {
+            List<String> dupErrorsFeatures = new ArrayList<>();
+            dupErrorsFeatures.addAll(Arrays.asList(error.split("\\r?\\n")));
+            failuresMap.put("duplicates", dupErrorsFeatures);
+        }
+        if (!failuresMap.isEmpty()) {
+            validateAnomalyDetectorResponse();
```

Comment:
naming it to `sendAnomalyDetectorValidationResponse()` makes more sense to me

Reply:
fixed